src/txn/txn_timestamp.c

```diff
@@ -668,6 +680,83 @@ __wt_txn_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
 	return (0);
 }
 
+/*
+ * __wt_txn_parse_prepare_timestamp --
+ *	Parse a request to set a transaction's prepare_timestamp.
+ */
+int
+__wt_txn_parse_prepare_timestamp(
+    WT_SESSION_IMPL *session, const char *cfg[], wt_timestamp_t *timestamp)
+{
+	WT_CONFIG_ITEM cval;
+
+	WT_RET(__wt_config_gets_def(session,
+	    cfg, "prepare_timestamp", 0, &cval));
+	if (cval.len > 0) {
+#ifdef HAVE_TIMESTAMPS
+		WT_TXN *prev;
+		WT_TXN_GLOBAL *txn_global;
+		wt_timestamp_t oldest_ts;
+		char hex_timestamp[2 * WT_TIMESTAMP_SIZE + 1];
+
+		txn_global = &S2C(session)->txn_global;
+
+		if (F_ISSET(&session->txn, WT_TXN_HAS_TS_COMMIT))
+			WT_RET_MSG(session, EINVAL,
+			    "commit timestamp should not have been set before "
+			    "prepare transaction");
+
```

Comment:
What if an application tries to set a prepare ts again for a ts, would we allow that?

Reply:
it will through error, please refer SESSION_API_CALL